app/services/complete_moab_handler.rb

```diff
@@ -247,14 +250,16 @@ def update_online_version(status = nil, set_status_to_unexp_version = false, che
   def update_cm_po_set_status
     if moab_validation_errors.empty?
       complete_moab.upd_audstamps_version_size(ran_moab_validation?, incoming_version, incoming_size)
-      pres_object.current_version = incoming_version
+      pres_object.current_version = incoming_version if checking_primary_moab? # we only want to track highest seen version based on primary
       pres_object.save!
     else
       update_status('invalid_moab')
     end
   end
 
   def raise_rollback_if_cm_po_version_mismatch
+    return unless checking_primary_moab?
```

Comment:
we're not so worried if a non-primary is not in sync with the parent `PreservedObject`, since only the primary is expected to be kept up to date automatically by pres robots.